vortex-tui/src/convert.rs

```diff
@@ -0,0 +1,128 @@
+#![allow(clippy::use_debug)]
+
+use std::path::Path;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Instant;
+
+use arrow_array::StructArray as ArrowStructArray;
+use futures_util::Stream;
+use indicatif::ProgressBar;
+use parquet::arrow::ParquetRecordBatchStreamBuilder;
+use pin_project::pin_project;
+use tokio::fs::File;
+use vortex::arrays::ChunkedArray;
+use vortex::arrow::FromArrowArray;
+use vortex::dtype::DType;
+use vortex::error::{VortexExpect, VortexResult};
+use vortex::file::VortexWriteOptions;
+use vortex::stream::{ArrayStream, ArrayStreamArrayExt};
+use vortex::{Array, ArrayRef};
+
+#[derive(Default)]
+pub struct Flags {
+    pub quiet: bool,
+}
+
+/// Convert Parquet files to Vortex.
+pub async fn exec_convert(input_path: impl AsRef<Path>, flags: Flags) -> VortexResult<()> {
+    if !flags.quiet {
+        println!(
+            "Converting input Parquet file: {}",
+            input_path.as_ref().display()
+        );
+    }
+
+    let wall_start = Instant::now();
+
+    let output_path = input_path.as_ref().with_extension("vortex");
+    let file = File::open(input_path).await?;
+    let mut reader = ParquetRecordBatchStreamBuilder::new(file).await?.build()?;
+    let mut chunks = Vec::new();
+
+    while let Some(mut reader) = reader.next_row_group().await? {
+        for batch in reader.by_ref() {
+            let batch = ArrowStructArray::from(batch?);
+            let next_chunk = ArrayRef::from_arrow(&batch, true);
+            chunks.push(next_chunk);
```

Comment:
You should be able to just stream these straight into the writer without buffering everything in memory. using the row group count as your progress indicator